### PR TITLE
Minor changes needed in API-design-guidelines.md for CloudEvents

### LINKED ISSUE #83: Consolidation of minor changes needed in design guidelines for CloudEvents content
**Problem description**
Certain minor points have been identified within the design guidelines that need to be changed for the CloudEvents content. This is a consolidation issue for the same. 

**Expected behavior**

**Alternative solution**

**Additional context**

### PR DESCRIPTION
**Pull request by rartych**
#### What type of PR is this?
* correction

#### What this PR does / why we need it:

Certain minor points have been identified within the design guidelines that need to be changed for the CloudEvents content.
Some editorial changes for better readability added.

#### Which issue(s) this PR fixes:

Fixes #83

#### Changelog input

```
Corrections for Event subscriptions and CloudEvents notifications implemented

```

### CODE REVIEW

**Review by bigludo7 — APPROVED** [review]
Look good for me 
Thanks !

**Review by patrice-conil — DISMISSED** [review]
/LGTM

**Review by shilpa-padgaonkar — APPROVED** [review]
/LGTM

**Review by eric-murray — APPROVED** [review]

File: documentation/API-design-guidelines.md [hunk 1]
```diff
@@ -1244,8 +1244,8 @@ In order to ease developer adoption, the pattern for Resource-based event subscr
 | ----- |	-----  |	 -----  | 
 | POST | `/subscriptions` |  Operation to request an event subscription.     |
```
Comment (bigludo7) [inline documentation/API-design-guidelines.md:1245]: Add sentence (proposal): 
As the subscription could be created synchronously or asynchronously both status code 201 and 202 must be described in the yaml.

File: documentation/API-design-guidelines.md [hunk 2]
```diff
@@ -1244,8 +1244,8 @@ In order to ease developer adoption, the pattern for Resource-based event subscr
...
 | GET | `/subscriptions` |  Operation to retrieve a list of event subscriptions - could be an empty list.  e.g. `GET /subscriptions?type=org.camaraproject.device-status.v1.roaming-status&expiresAt.lt=2023-03-17` |
-| GET | `/subscriptions/{subscriptionsId}` | Operation to retrieve an event subscription |
-| DELETE | `/subscriptions/{subscriptionsId}` | Operation to delete an event subscription |
+| GET | `/subscriptions/{subscriptionId}` | Operation to retrieve an event subscription |
```
Comment (bigludo7) [inline documentation/API-design-guidelines.md:1247]: Add sentence (proposal):
Note: If the GET `/subscriptions/{subscriptionId}` is not able to retrieve a recently created subscription in async mode, a 404 code is sent back.

File: documentation/API-design-guidelines.md [hunk 3]
```diff
@@ -1244,8 +1244,8 @@ In order to ease developer adoption, the pattern for Resource-based event subscr
...
+| DELETE | `/subscriptions/{subscriptionId}` | Operation to delete an event subscription |
```
Comment (bigludo7) [inline documentation/API-design-guidelines.md:1248]: Add sentence (proposal):
As the subscription deletion could be handled synchronously or asynchronously both status code 202 and 204 must be described in the yaml.

**Review by bigludo7 — COMMENTED** [review]
@rartych as discussed today 3 sentences proposal for the subscription operation table. I was not able to edit the PR but just comment.

Seems that our discussion was aligned with @PedroDiez thought.

**Comment by rartych** [comment]
@RubenBG7 @jordonezlucena Please review this PR as code owners.

**Review by bigludo7 — APPROVED** [review]
LGTM

**Review by jordonezlucena — APPROVED** [review]
LGTM

**Review by shilpa-padgaonkar — APPROVED** [review]
/LGTM

**Review by eric-murray — APPROVED** [review]

**Pull request merged**